LuaMenu/widgets/garbage_collection.lua

```diff
@@ -0,0 +1,35 @@
+
+function widget:GetInfo()
+	return {
+		name = "Garbage Collection",
+		desc = "Periodically forces garbage collection" ,
+		layer = -math.huge,
+		enabled   = true,
+	}
+end
+
+
+local basememlimit = 200000
+local garbagelimit = basememlimit -- in kilobytes, will adjust upwards as needed
+local lastGCchecktime = Spring.GetTimer()
+
+function widget:Update()
+    if Spring.DiffTimers(Spring.GetTimer(), lastGCchecktime) > 1 then
+		lastGCchecktime = Spring.GetTimer()
+		local ramuse = gcinfo()
+		--Spring.Echo("RAMUSE",ramuse)
+		if ramuse > garbagelimit then 
+			collectgarbage("collect")
+			collectgarbage("collect")
```

Comment:
This needs an explanatory comment of why `collectgarbage` needs to be called twice.

Reply:
mark + sweep